backend/app/controllers/spree/admin/promotion_code_batches_controller.rb

```diff
@@ -0,0 +1,27 @@
+module Spree
+  module Admin
+    class PromotionCodeBatchesController < ResourceController
+      belongs_to 'spree/promotion'
+
+      create.after :build_promotion_code_batch
+
+      def download
+        require 'csv'
+        @promotion_code_batch = Spree::PromotionCodeBatch.find_by_id(params[:promotion_code_batch_id])
```

Comment:
Nitpick: Might as well use `find` here, rather than `find_by_id` so that it raises with a clear error message if the id is invalid?